src/test/java/uk/gov/justice/digital/delius/service/NsiServiceTest.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.justice.digital.delius.service;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.digital.delius.data.api.Nsi;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Event;
+import uk.gov.justice.digital.delius.jpa.standard.repository.NsiRepository;
+import uk.gov.justice.digital.delius.transformers.NsiTransformer;
+
+@ExtendWith(MockitoExtension.class)
+public class NsiServiceTest {
+
+    private static final Long OFFENDER_ID = 123L;
+    private static final Long EVENT_ID = 124L;
+    private static final Event EVENT = Event.builder().softDeleted(0L).build();
+
+    @Mock
+    private NsiRepository nsiRepository;
+
+    @Mock
+    private NsiTransformer nsiTransformer;
+
+    @InjectMocks
+    private NsiService nsiService;
+
+    @DisplayName("All NSIs are fetched and returned, having been transformed")
+    @Test
+    void whenFetchNsisNoneFilterByCodeOrSoftDeleted() {
+
+        final uk.gov.justice.digital.delius.jpa.standard.entity.Nsi nsiEntity = buildNsi(EVENT, "BRE");
```

Comment:
very minor, but maybe use `var` instead of the types for readability

Reply:
Will do. Keep forgetting I am in Java 11 land :-)